internal/backend/runtime/omni/migration/migration_test.go

```diff
@@ -1798,6 +1798,138 @@ func (suite *MigrationSuite) TestMoveEtcdBackupStatuses() {
+  features:
+    foo: bar
+    apidCheckExtKeyUsage: true
+    baz: qux
+    diskQuotaSupport: true
+    quux: corge
+`)))
+
+	// patch expected: **only** diskQuotaSupport needs to be preserved
+	cluster14config1 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-1")
+	cluster14config1.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config1.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+	foo: bar
+    diskQuotaSupport: true
+    apidCheckExtKeyUsage: true
+`)))
+
+	// no patch expected: it does not have any of the features to be preserved
+	cluster14config2 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-2")
+	cluster14config2.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config2.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+	foo: bar
+`)))
+
+	// no patch expected, since it will be in tearing down phase
+	cluster14config3 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-3")
+	cluster14config3.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config1.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+	foo: bar
+    diskQuotaSupport: true
+`)))
+
+	suite.Require().NoError(suite.state.Create(ctx, clusterConfigVersion12))
+	suite.Require().NoError(suite.state.Create(ctx, clusterConfigVersion14))
+	suite.Require().NoError(suite.state.Create(ctx, cluster12config1))
+	suite.Require().NoError(suite.state.Create(ctx, cluster12config2))
+	suite.Require().NoError(suite.state.Create(ctx, cluster12config3))
+	suite.Require().NoError(suite.state.Create(ctx, cluster14config1))
+	suite.Require().NoError(suite.state.Create(ctx, cluster14config2))
+	suite.Require().NoError(suite.state.Create(ctx, cluster14config3))
+
+	_, err := suite.state.Teardown(ctx, cluster14config3.Metadata())
+	suite.Require().NoError(err)
+
+	suite.Require().NoError(suite.manager.Run(ctx, migration.WithFilter(filterWith("createVersionContractRevertConfigPatch"))))
+
+	patchList, err := safe.StateListAll[*omni.ConfigPatch](ctx, suite.state)
+	suite.Require().NoError(err)
+
+	suite.Require().Equal(3, patchList.Len(), "there should be 3 config patches")
+
+	assertPatch := func(machine *omni.ClusterMachineConfig, expected string) {
+		for patch := range patchList.All() {
+			cm, _ := patch.Metadata().Labels().Get(omni.LabelClusterMachine)
+			if cm == machine.Metadata().ID() {
+				suite.True(strings.HasPrefix(patch.Metadata().ID(), "000-"))
+				suite.True(strings.HasSuffix(patch.Metadata().ID(), "-preserve-version-contract"))
+
+				buf, err := patch.TypedSpec().Value.GetUncompressedData()
+				suite.Require().NoError(err)
+
+				dataStr := string(buf.Data())
+
+				var m map[string]any
+
+				// assert that it is a valid yaml
+				suite.Require().NoError(yaml.Unmarshal([]byte(dataStr), &m))
```

Comment:
I'd suggest to decode the file using talos config parser.

Reply:
Changed it to use `omni.ValidateConfigPatch(...)`, which does exactly that under the hood.